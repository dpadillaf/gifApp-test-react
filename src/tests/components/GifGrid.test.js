import React from 'react';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock( '../../hooks/useFetchGifs' );


describe('test in GifGrid', () => {
    
    test('debe mostrarse correctamenyte', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        let wrapper = shallow( <GifGrid category={ 'snk' } /> );
        expect( wrapper ).toMatchSnapshot();

    })
    
    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [ {
            id: 'abc',
            title: 'un title',
            url: 'https://google.com'
        },
        {
            id: 'cba',
            title: 'un title 2',
            url: 'https://google.com'
        } ];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        let wrapper = shallow( <GifGrid category={ 'snk' } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length );

    })
    

})
