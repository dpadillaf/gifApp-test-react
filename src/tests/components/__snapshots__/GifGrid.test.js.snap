// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGrid debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__heartBeat"
  >
    snk
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="un title"
      url="https://google.com"
    />
    <GifGridItem
      id="cba"
      key="cba"
      title="un title 2"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`test in GifGrid debe mostrarse correctamenyte 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__heartBeat"
  >
    snk
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
